### PR TITLE
Update the Org Repository Policy per the 2022 PSC

### PR DESCRIPTION
**Pull request by gpshead**
This is a result of the discussion started by the 2022 Python Steering Council on https://discuss.python.org/t/new-python-organization-repository-policy/17376 and
https://mail.python.org/archives/list/python-dev@python.org/message/FAASOMUNQZDJBCOTZK6I2KH55SDOONF6/.

Petr's text from the discuss thread is used, omitting any mention of the PSF Contributor Agreement (CLA) as discussion revealed that it may not make sense for all repos (see thread).

It also adopts a suggestion from Joannah to have people discuss additions in our Committers forum before bringing an addition to the Python Steering Council.

### CODE REVIEW

**Comment by gpshead** [comment]
this will show up in https://devguide.python.org/developer-workflow/development-cycle/

File: developer-workflow/development-cycle.rst [hunk 1]
```diff
@@ -216,16 +216,43 @@ The source code is currently hosted on `GitHub
...
+accounts or other GitHub orgs. It is relatively easy to move a repository to
+the organization once it is mature. For example, this would now apply to
+experimental features like `asyncio <https://github.com/python/asyncio>`_,
+`exceptiongroups <https://github.com/python/exceptiongroups>`_, or
+`typed_ast <https://github.com/python/typed_ast>`_, and drafts of new guides
+and other documentation (e.g. `redistributor-guide <https://github.com/python/redistributor-guide>`_).
+
+General-use tools and libraries (e.g. `mypy <https://github.com/python/mypy>`_
+or `black <https://github.com/psf/black>`_) should also be developed outside
+the python organization, unless core devs (as represented by the SC)
```
Comment (hugovk) [inline developer-workflow/development-cycle.rst:250]:
```suggestion
or `Black <https://github.com/psf/black>`_) should also be developed outside
the Python organization, unless core devs (as represented by the SC)
```

**Comment by hugovk** [comment]
Build preview here: https://cpython-devguide--1036.org.readthedocs.build/developer-workflow/development-cycle/#organization-repository-policy

File: developer-workflow/development-cycle.rst [hunk 2]
```diff
@@ -216,16 +216,43 @@ The source code is currently hosted on `GitHub
...
+* Organization-related repositories (e.g. the `Code of Conduct <https://github.com/python/pycon-code-of-conduct>`_, `.github <https://github.com/python/.github>`_)
+* Documentation and websites for all the above (e.g. `python.org repository <https://github.com/python/pythondotorg>`_, `PEPs <https://github.com/python/peps>`_, `Devguide <https://github.com/python/devguide>`_, docs translations)
+* Infrastructure for all the above (e.g. `docsbuild-scripts <https://github.com/python/docsbuild-scripts>`_, `buildmaster-config <https://github.com/python/buildmaster-config>`_)
+* Discussions and notes around official development-related processes and events (e.g. `steering-council <https://github.com/python/steering-council>`_, `core-sprint <https://github.com/python/core-sprint>`_)
+
+Before adding a new repository to the organization, have discussion to seeking
+consensus in our `Committers forum <https://discuss.python.org/c/committers/5>`_.
+Once people are satisfied with that, ask the `Python steering council <https://github.com/python/steering-council>`_
+to grant permission.
```
[the remark this answers is not in the document]
Comment (CAM-Gerlach) [reply developer-workflow/development-cycle.rst:235]: Both of these are ungrammatical for multiple reasons; here's an alternative that is (there are number of other options, too):

```suggestion
Before adding a new repository to the organization, open a discussion to seek consensus
in the `Committers Discourse category <https://discuss.python.org/c/committers/5>`_.
Once people are satisfied with that, ask the `Python steering council <https://github.com/python/steering-council>`_
to grant permission.
```

File: developer-workflow/development-cycle.rst (hunk 1 above)
Comment (CAM-Gerlach) [reply developer-workflow/development-cycle.rst:250]: or

```suggestion
or `black <https://github.com/psf/black>`_) should also be developed outside
the ``python`` organization, unless core devs (as represented by the SC)
```

**Comment by encukou** [comment]
Thanks for getting around to this!
Looks good to me, modulo the editorial fixes suggested above.

File: developer-workflow/development-cycle.rst [hunk 3]
```diff
@@ -216,16 +216,43 @@ The source code is currently hosted on `GitHub
...
+
+Note that several repositories remain in the organization for historic reasons,
+and would probably not be appropriate to add today.
+
+Generally, new repositories should start their life under personal GitHub
+accounts or other GitHub orgs. It is relatively easy to move a repository to
+the organization once it is mature. For example, this would now apply to
+experimental features like `asyncio <https://github.com/python/asyncio>`_,
+`exceptiongroups <https://github.com/python/exceptiongroups>`_, or
+`typed_ast <https://github.com/python/typed_ast>`_, and drafts of new guides
```
Comment (JelleZijlstra) [inline developer-workflow/development-cycle.rst:245]: I'm not sure this is quite historically accurate. `typed-ast` was originally created to support typecheckers, similar to typeshed, without intention to move it to CPython.

Comment (gpshead) [reply developer-workflow/development-cycle.rst]: thanks, we're listing more than is needed to make the point. I just took typed_ast out of the list.

**Review by JelleZijlstra — APPROVED** [review]

**Review by brettcannon — APPROVED** [review]

**Review by CAM-Gerlach — APPROVED** [review]

**Pull request merged**